daemon/commit.go

```diff
@@ -42,8 +42,10 @@ func (daemon *Daemon) ContainerCommit(job *engine.Job) engine.Status {
 // The image can optionally be tagged into a repository
 func (daemon *Daemon) Commit(container *Container, repository, tag, comment, author string, pause bool, config *runconfig.Config) (*image.Image, error) {
 	if pause {
-		container.Pause()
-		defer container.Unpause()
+		if ispaused := container.IsPaused(); !ispaused {
+			container.Pause()
```

Comment:
shouldn't this be outside of the if-statement since we want to pause it irrespective of whether its currently paused, no?